### PR TITLE
Feat: add survey list page with mock api

### LINKED ISSUE #35: [P0] Survey list page with temp data
See

- [fake api](https://jsonplaceholder.typicode.com/)
- [mock api](https://mockapi.io/)
  - https://6406918177c1a905a0ddbe34.mockapi.io/api/survey-lists
- Postman (After set ERD...)

### PR DESCRIPTION
**Pull request by sanghyun1128**
1. Survey LIst page
- When width is not enough, it's hard to show all of 'title', 'required_auth' and 'ended_date'...  so depending on the width, 'ended_date' row was made invisible

2. Api
- install [axios](https://axios-http.com/kr/) pakage
- make [axios instance](https://axios-http.com/kr/docs/instance) and api endpoints constant for prevent repetition

This PR closes #35

### CODE REVIEW

File: web/src/api/axios.ts [hunk 1]
```diff
@@ -0,0 +1,33 @@
+import axios, { AxiosInstance } from 'axios';
+
+const instance: AxiosInstance = axios.create({
```
Comment (seiwonpark) [inline web/src/api/axios.ts:3]: Axios instance! Awesome!

File: web/src/api/axios.ts [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+  baseURL: 'https://capstone-mock-api.fly.dev',
+  // timeout: 15000,
```
Comment (seiwonpark) [inline web/src/api/axios.ts:5]: FYI. Timeout is good but we need to add `AbortController` later. It's okay for now.

File: web/src/api/axios.ts [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+});
+
+// 요청 인터셉터 추가하기
+instance.interceptors.request.use(
```
Comment (seiwonpark) [inline web/src/api/axios.ts:9]: So have you tried this out? Wrapping `axios.get()` with try catch and which acts first? If you know about the order of precedence, please share it!

File: web/src/api/axios.ts [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+    return config;
+  },
+  (error) => {
+    // 요청 오류가 있는 작업 수행
+    return Promise.reject(error);
+  }
+);
+
+// 응답 인터셉터 추가하기
+instance.interceptors.response.use(
```
Comment (seiwonpark) [inline web/src/api/axios.ts:21]: Same above. If you've tried this, is there any priority for orders like `try catch` and `interceptors.response`?

File: web/src/api/request.ts [hunk 5]
```diff
@@ -0,0 +1,14 @@
+interface ApiRequests {
+  healthCheck: string;
+  fetchSurveyListAll: string;
+  fetchSurvey: string;
+  fetchSurveyListPage: string;
+}
```
Comment (seiwonpark) [inline web/src/api/request.ts:6]: Could you add a blank line here..? I know it's not a big deal but for consistency of our entire code.

File: web/src/routes/SurveyPages/SurveyListPage.tsx [hunk 6]
```diff
@@ -1,5 +1,201 @@
...
+  padding: 2vh;
+  font-size: 2vh;
+  font-weight: bold;
+  text-align: center;
+  color: ${(props) => props.theme.colors.default};
+`;
+
+const Title = styled(Item)`
+  min-width: 15vh;
+  flex: 1;
```
Comment (seiwonpark) [inline web/src/routes/SurveyPages/SurveyListPage.tsx:56]: Reason for Style.

Could you add a blank line here?

File: web/src/routes/SurveyPages/SurveyListPage.tsx [hunk 7]
```diff
@@ -1,5 +1,201 @@
...
+
+const Authlist = styled(Item)`
+  min-width: 160px;
+  width: 20vw;
+`;
+
+const EndDate = styled(Item)`
+  min-width: 150px;
+  width: 13vw;
+  text-align: center;
```
Comment (seiwonpark) [inline web/src/routes/SurveyPages/SurveyListPage.tsx:71]: Reason for Style.

Could you add a blank line here?

File: web/src/routes/SurveyPages/SurveyListPage.tsx [hunk 8]
```diff
@@ -1,5 +1,201 @@
...
+import Header from '../../components/Header';
+import { useTheme } from '../../hooks/useTheme';
+
+const Container = styled.div`
+  width: 100vw;
+  height: 100vh;
+  background-color: ${(props) => props.theme.colors.container};
+`;
+
+const ListContainer = styled.div`
```
Comment (seiwonpark) [inline web/src/routes/SurveyPages/SurveyListPage.tsx]: So, looks like this is a table. Avoid using all tags with `<div>`s. Select proper tags for its component!

`ListContainer` should be more like `<table>`. See https://developer.mozilla.org/en-US/docs/Web/HTML/Element/table

File: web/src/routes/SurveyPages/SurveyListPage.tsx [hunk 9]
```diff
@@ -1,5 +1,201 @@
...
+  width: 90vw;
+  height: 94vh;
+  display: flex;
+  flex-direction: column;
+  padding: 3vh 5vw 3vh 5vw;
+  background-color: ${(props) => props.theme.colors.container};
+`;
+
+const ListColumn = styled.div`
```
Comment (seiwonpark) [inline web/src/routes/SurveyPages/SurveyListPage.tsx]: Then this should be `<thead>`. See https://developer.mozilla.org/en-US/docs/Web/HTML/Element/table

File: web/src/routes/SurveyPages/SurveyListPage.tsx [hunk 10]
```diff
@@ -1,5 +1,201 @@
...
+  display: flex;
+  flex-direction: row;
+  justify-content: center;
+`;
+
+const Item = styled.div`
```
Comment (seiwonpark) [inline web/src/routes/SurveyPages/SurveyListPage.tsx]: This should be `td`. See https://developer.mozilla.org/en-US/docs/Web/HTML/Element/table

And for other tags, please change or add proper tags!

File: web/src/routes/SurveyPages/SurveyListPage.tsx [hunk 11]
```diff
@@ -1,5 +1,201 @@
...
+const DriverLicense = styled(Auth)`
+  background-color: #34a853;
+`;
+const AuthNone = styled(Auth)`
+  background-color: #b3b2b2;
+`;
+
+const AuthSummary = styled.span`
+  font-size: 13px;
+`;
```
Comment (seiwonpark) [inline web/src/routes/SurveyPages/SurveyListPage.tsx:131]: Reason for Style.

Could you add a blank line here?

File: web/src/routes/SurveyPages/SurveyListPage.tsx [hunk 12]
```diff
@@ -1,5 +1,201 @@
...
+  required_authentications: Array<string>;
+}
 
 export default function SurveyListPage() {
-  return <div>SurveyListPage</div>;
+  const [theme, toggleTheme] = useTheme();
+  const [surveys, setSurveys] = useState<Survey[]>([]);
+  const navigate = useNavigate();
+
+  const fetchSurveyData = async (): Promise<AxiosResponse<Survey[]>> => {
```
Comment (seiwonpark) [inline web/src/routes/SurveyPages/SurveyListPage.tsx]: You're already updating states with `setSurveys` and if you see `useEffect` hook, you don't actually need to return the results. `Promise<void>` should work.

Or is there any other reason for this?

File: web/src/routes/SurveyPages/SurveyListPage.tsx [hunk 13]
```diff
@@ -1,5 +1,201 @@
...
+      case 'ID':
+        return <Id key={auth}>신분증</Id>;
+      case 'MOBILE_PHONE':
+        return <MobilePhone key={auth}>휴대폰인증</MobilePhone>;
+      case 'DRIVER_LICENSE':
+        return <DriverLicense key={auth}>운전면허</DriverLicense>;
+      default:
+        return <AuthNone key={auth}>Undefined</AuthNone>;
+    }
+  };
```
Comment (seiwonpark) [inline web/src/routes/SurveyPages/SurveyListPage.tsx:170]: Reason for Style.

Could you add a blank line here?

File: web/src/routes/SurveyPages/SurveyListPage.tsx [hunk 14]
```diff
@@ -1,5 +1,201 @@
...
+    <Container theme={theme}>
+      <Header theme={theme} toggleTheme={toggleTheme} />
+      <ListContainer theme={theme}>
+        <ListColumn>
+          <HeadTitle theme={theme}>설문 제목</HeadTitle>
+          <HeadAuthList theme={theme}>필수인증</HeadAuthList>
+          <HeadEndDate theme={theme}>설문 종료일</HeadEndDate>
+        </ListColumn>
+
+        {surveys.slice(0, 8).map((survey) => (
```
Comment (seiwonpark) [inline web/src/routes/SurveyPages/SurveyListPage.tsx]: Oh you're slicing it! Do you want it to be 8 rows for each page?

File: web/src/routes/SurveyPages/SurveyListPage.tsx [hunk 15]
```diff
@@ -1,5 +1,201 @@
...
+          <ListColumn key={survey.survey_id} theme={theme}>
+            <Title onClick={() => navigate(`/survey/${survey.survey_id}`)} theme={theme}>
+              {survey.title}
+            </Title>
+            <Authlist theme={theme}>
+              {survey.required_authentications.length === 0 ? (
+                <AuthNone>제한없음</AuthNone>
```
Comment (seiwonpark) [inline web/src/routes/SurveyPages/SurveyListPage.tsx]: Use `makeAuthLabel('')` and modify `default` value to `제한없음`. Avoid hard-coded one.

**Review by seiwonpark — CHANGES_REQUESTED** [review]
Please update the code and re-request! Impressed with Axios instance. Cool!

File: web/src/api/axios.ts (hunk 4 above)
Comment (seiwonpark) [reply web/src/api/axios.ts:21]: Q. Does the option works together? No? Then which one?

File: web/src/routes/SurveyPages/SurveyListPage.tsx (hunk 14 above)
Comment (seiwonpark) [inline web/src/routes/SurveyPages/SurveyListPage.tsx]: Could you add a component like `No data` when there's no survey list?

File: web/src/api/axios.ts (hunk 4 above)
Comment (sanghyun1128) [reply web/src/api/axios.ts:21]: `interceptor` works first. if error occured, `interceptor` return `Promise.reject(error)` then `catch` can catch error.
But if `interceptor` return `Promise<any>` then `catch` can't catch error.

So we can use `interceptor` for repeatitive error handle, and also `catch` if needed.

File: web/src/routes/SurveyPages/SurveyListPage.tsx (hunk 12 above)
Comment (sanghyun1128) [reply web/src/routes/SurveyPages/SurveyListPage.tsx]: No.. I was told I had to return the `Promise` object, so I did. I didn't know `Promise<void>` existed.

File: web/src/routes/SurveyPages/SurveyListPage.tsx (hunk 14 above)
Comment (sanghyun1128) [reply web/src/routes/SurveyPages/SurveyListPage.tsx]: It got too long on the mobile, so I cut it for now.
I'm not sure how many would be good.

Comment (sanghyun1128) [reply web/src/routes/SurveyPages/SurveyListPage.tsx]: Ok! I'll add with pagination.

File: web/src/api/axios.ts (hunk 4 above)
Comment (seiwonpark) [reply web/src/api/axios.ts:21]: Very nice! So we can use both `try catch` and `interceptor` together. Thank you!

File: web/src/routes/SurveyPages/SurveyListPage.tsx (hunk 8 above)
Comment (seiwonpark) [reply web/src/routes/SurveyPages/SurveyListPage.tsx]: @sanghyun1128  Please resolve this first

File: web/src/routes/SurveyPages/SurveyListPage.tsx (hunk 14 above)
Comment (seiwonpark) [reply web/src/routes/SurveyPages/SurveyListPage.tsx]: Ok then let it be 8 then. I'll fix it.

**Review by seiwonpark — CHANGES_REQUESTED** [review]
Please check unresolved requests.

**Review by seiwonpark — APPROVED** [review]
LGTM. Thanks!

**Pull request merged**